hollow/src/main/java/com/netflix/hollow/api/consumer/index/SelectFieldPathResultExtractor.java

```diff
@@ -71,7 +72,7 @@ static IllegalArgumentException incompatibleSelectType(Class<?> selectType, Stri
     static <T> SelectFieldPathResultExtractor<T> from(
             Class<? extends HollowAPI> apiType, HollowDataset dataset, Class<?> rootType, String fieldPath,
             Class<T> selectType) {
-        String rootTypeName = rootType.getSimpleName();
+        String rootTypeName = HollowObjectTypeMapper.getDefaultTypeName(rootType);
```

Comment:
I had some trouble testing the changes in this file, Paul's eyes would be useful here

Reply:
Here's an example:
```
HashIndexSelect<Movie, Actor, String> index = HashIndex.from(consumer, Movie.class)
    .selectField("cast.element.actor", Actor.class)
    .usingPath("title.value", String.class);

String title = "The Imitation Game";
Stream<Actor> m = index.findMatch(title);  
```
It's a bit like:
```
SELECT Actor FROM Movie
WHERE title='The Imitation Game';
```
So you need to add another annotated test class for selection type referenced from the top type.